### ISSUE TITLE
`addMark()` doesn't work?

### ISSUE THREAD

**Issue by alphonso06** [victorsoares96/epubjs-react-native#117]
Tried using `addMark()` to test out highlighting via CFI Range, but it didn't seem to work for `onReady`, `onStarted`, and even `onRendered`.

Tested if the `onPress()` was working and by adding a simple `console.info("book pressed")` I was able to confirm that it was working. So that's where I decided to put the `addMark()` hook instead.

Given the following page, this will start you at a specific page. The CFI Range I provided for `addMark()` should highlight the word "Niagara", but it doesn't.

```ts
export default function Page() {
    const { width, height } = useWindowDimensions()
    const { addMark } = useReader()
    
    return (
        <SafeAreaView>
            <ReaderProvider>
                <Reader
                    src="https://s3.amazonaws.com/moby-dick/OPS/package.opf"
                    width={width}
                    height={height}
                    fileSystem={useFileSystem}
                    initialLocation="epubcfi(/6/14!/4/2/14/2[c001p0006]/1:348)"
                    onPress={() => addMark("highlight", "epubcfi(/6/14!/4/2/14/2[c001p0006],/1:911,/1:918)")}
                />
            </ReaderProvider>
        </SafeAreaView>
    )
}
```

Any ideas how to use this properly? What am I missing here? 🤔

--edit--

I tested this on Expo Go (SDK Version 49), with an Android 13 device.

**Comment by victorsoares96** [comment]
Hello @alphonso06, did @Goldlightdrake suggestion help you with the problem?

**Comment by smallbonelu** [comment]
It seems that all the values get from the useReader hook is still the default value. Which cause all methods from useReader hook are not working as expected. 

Here is the result from React Dev Tool:
The useReader hook get the default value.
<img width="1025" alt="image" src="https://github.com/victorsoares96/epubjs-react-native/assets/15611493/613078e1-ffa3-4d01-a911-14ec99475d26">
But the Context Provider's value is the latest
<img width="1023" alt="image" src="https://github.com/victorsoares96/epubjs-react-native/assets/15611493/42f4af47-752e-4445-ae69-8bfdcc9c9b6e">

**Comment by victorsoares96** [comment]
Hi everyone, I finally fixed this problem, sorry for the delay. The fix will soon be available in version `1.4.0`
